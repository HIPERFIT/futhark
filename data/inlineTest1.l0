fun int fun1(int a, int b) = a + b

fun int fun2(int a, int b) = fun1(a,b) * (a+b)

fun int fun3(int a, int b) = fun2(a,b) + a + b

fun int get10() = read(int)

fun int main() = 
    let n = get10() in
    let m = get10() in
      fun1(n,m) + fun2(n+n,m+m) + fun3(3*n,3*m) + fun2(2,n) + fun3(n,3)



