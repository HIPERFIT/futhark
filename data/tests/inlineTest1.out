Result of evaluation: 24730855
